app/client/ui/DocumentSettings.ts

```diff
@@ -350,13 +352,76 @@ export class DocSettingsPage extends Disposable {
           ]),
         ]),
       ),
+      this._buildAttachmentUploadSection(),
     ]);
   }
 
+  private _buildAttachmentUploadSection() {
+    const isUploadingObs = Observable.create(this, false);
+    const buttonText = Computed.create(this, (use) => use(isUploadingObs) ? t('Uploading...') : t('Upload'));
+
+    const uploadButton = cssSmallButton(
+      dom.text(buttonText),
+      dom.on('click',
+        async () => {
+          // This may never return due to openFilePicker. Anything past this point isn't guaranteed
+          // to execute.
+          const file = await this._pickAttachmentsFile();
+          if (!file) {
+            return;
+          }
+          isUploadingObs.set(true);
```

Comment:
In the presence of the comment above, I think it is better to test if the owner is not disposed after promise returns? (Here and below).

Reply:
I'm not sure what you mean, can you describe the change in more detail? 

Are you suggesting we check if the owner is disposed before setting the observable? 

(Also - I just had a look at Observable. It's looks like you can still trigger change callbacks after it's disposed, and that's *weird*)

Reply:
You are doing an async operation. By the time the promise is resolved, someone could move to another page, destroying all observables you are setting.

Reply:
I've pushed some guards for this. Is this something that Observables should be handling internally?

It feels like once disposed, they should either error or do nothing if `set` is called on them.

Reply:
Let me know if the additional checks are sufficient!